connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2385,6 +2385,13 @@ class SparkConnectPlanner(val session: SparkSession) {
       writer.queryName(writeOp.getQueryName)
     }
 
+    if (writeOp.hasForeachWriter) {
+      val foreach = writeOp.getForeachWriter.getPythonWriter
+      val pythonFcn = transformPythonFunction(foreach)
+      writer.foreachConnect(
+        new PythonForeachWriter(pythonFcn, dataset.schema).asInstanceOf[ForeachWriter[Any]])
```

Comment:
Do we need the cast here? `PythonForeachWriter` already extends ForeachWriter.

Reply:
Yes we do need it:
```
[error] /home/wei.liu/oss-spark/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:2392:9: type mismatch;
[error]  found   : org.apache.spark.sql.execution.python.PythonForeachWriter
[error]  required: org.apache.spark.sql.ForeachWriter[Any]
```